applets/head/head.go

```diff
@@ -94,10 +98,63 @@ func linesHead(w io.Writer, f *os.File, n int) error {
 	return s.Err()
 }
 
+func negativeLinesHead(w io.Writer, f *os.File, n int) error {
+	n *= -1
+	lastNLines := make([]string, n)
+	pos := 0
+
+	s := bufio.NewScanner(f)
+	for count := 0; s.Scan(); count++ {
+		if count >= n {
+			if _, err := fmt.Fprintln(w, lastNLines[pos]); err != nil {
+				return err
+			}
+		}
+		lastNLines[pos] = s.Text()
+		pos++
+		if pos >= n {
+			pos %= n
+		}
+	}
+	return s.Err()
+}
+
 func bytesHead(w io.Writer, f *os.File, n int64) error {
 	lr := io.LimitReader(f, n)
 	if _, err := io.Copy(w, lr); err != nil {
 		return err
 	}
 	return nil
 }
+
+func negativeBytesHead(w io.Writer, f *os.File, n int64) error {
+	n *= -1
+	lastNBytes := make([]byte, n)
```

Comment:
Thank you for adding new feature. But is it works? Could you add some tests?

Reply:
Done